### PR TITLE
[Merged by Bors] - feat: define `cfcHomSuperset`

### PR DESCRIPTION
**Pull request by j-loreaux**
This defines, for convenience, a map `cfcHomSuperset : C(s, R) → A` whenever `spectrum R a ⊆ s` in the natural way (i.e., by precomposition with the inclusion map).

---

This was useful for me to show that the decomposition of a selfadjoint `a = b - c` into a difference of nonnegative elements with `b * c = 0` is unique in #18138.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/NonUnital.lean
Comment (j-loreaux) [reply Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/NonUnital.lean:672]: it has a different meaning from Lean 3. In Lean 3 it reset the instance cache since that was frozen. In Lean 4, we don't freeze the instance cache anymore, so that's not relevant, but now the `I` means "inline". I thought it would be better to inline this, but maybe it doesn't matter. I'll check before merging.

Comment (dupuisf) [reply Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/NonUnital.lean:672]: Ah I see! Yeah I forgot what it did in Lean 4; inlining certainly makes sense here.

**Pull request closed without merging**
